src/Agent.PluginCore/PluginUtil.cs

```diff
@@ -0,0 +1,1611 @@
+                dir.Delete();
+            }
+        }
+
+        public static void DeleteFile(string path)
+        {
+            PluginUtil.NotNullOrEmpty(path, nameof(path));
+            var file = new FileInfo(path);
+            if (file.Exists)
+            {
+                RemoveReadOnly(file);
+                file.Delete();
+            }
+        }
+
+        /// <summary>
+        /// Recursively enumerates a directory without following directory reparse points.
+        /// </summary>
+        private static IEnumerable<FileSystemInfo> Enumerate(DirectoryInfo directory, CancellationTokenSource tokenSource)
+        {
+            PluginUtil.NotNull(directory, nameof(directory));
+            PluginUtil.Equal(false, directory.Attributes.HasFlag(FileAttributes.ReparsePoint), nameof(directory.Attributes.HasFlag));
+
+            // Push the directory onto the processing stack.
+            var directories = new Stack<DirectoryInfo>(new[] { directory });
+            while (directories.Count > 0)
+            {
+                // Pop the next directory.
+                directory = directories.Pop();
+                foreach (FileSystemInfo item in directory.GetFileSystemInfos())
+                {
+                    // Push non-reparse-point directories onto the processing stack.
+                    directory = item as DirectoryInfo;
+                    if (directory != null &&
+                        !item.Attributes.HasFlag(FileAttributes.ReparsePoint))
+                    {
+                        directories.Push(directory);
+                    }
+
+                    // Then yield the directory. Otherwise there is a race condition when this method attempts to initialize
+                    // the Attributes and the caller is deleting the reparse point in parallel (FileNotFoundException).
+                    yield return item;
+                }
+            }
+        }
+
+        private static void RemoveReadOnly(FileSystemInfo item)
+        {
+            PluginUtil.NotNull(item, nameof(item));
+            if (item.Attributes.HasFlag(FileAttributes.ReadOnly))
+            {
+                item.Attributes = item.Attributes & ~FileAttributes.ReadOnly;
+            }
+        }
+        #endregion
+
+        #region VarUtil
+        public static string PrependPath(string path, string currentPath)
+        {
+            PluginUtil.NotNullOrEmpty(path, nameof(path));
+            if (string.IsNullOrEmpty(currentPath))
+            {
+                // Careful not to add a trailing separator if the PATH is empty.
+                // On OSX/Linux, a trailing separator indicates that "current directory"
+                // is added to the PATH, which is considered a security risk.
+                return path;
+            }
+
+            return path + Path.PathSeparator + currentPath;
+        }
+
+        public static void PrependPath(string directory)
```

Comment:
doesnt the git source provider do this? does it an instruction for the agent to prepend path now?

Reply:
(for the benefit of downstream tasks)

Reply:
yes, you are right! i will change it to write ##vso[task.prepandpath]

Reply:
fixed.